### PR TITLE
Add regression coverage for float64 softmax tail gradients

### LINKED ISSUE #126633: tf.nn.softmax loses a finite first-derivative signal for float64 input
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

tf 2.21.0

### Custom code

Yes

### OS platform and distribution

Linux Ubuntu 22.04

### Mobile device

_No response_

### Python version

Python 3.13

### Bazel version

_No response_

### GCC/compiler version

_No response_

### CUDA/cuDNN version

_No response_

### GPU model and memory

_No response_

### Current behavior?

`tf.nn.softmax` returns zero for one first-derivative component whose correctly rounded `float64` reference value is finite, normal, and nonzero.

Softmax is invariant to adding the same constant to every logit, so every input VJP should sum to zero. In the reproducer below, the expected gradient is approximately `[5.551115123125775e-17, -5.551115123125775e-17]`.

However, TensorFlow returns `[0.0, -5.551115123125776e-17]`. The first gradient component is incorrectly collapsed to zero, and the resulting VJP no longer preserves the expected zero-sum relation.

Expected behavior?

The first-order autodiff result for `tf.nn.softmax` should preserve the finite, normal, nonzero `float64` gradient component. For the given input and cotangent, the gradient should be close to `[5.551115123125775e-17, -5.551115123125775e-17]`, and the VJP should sum to zero.

### Standalone code to reproduce the issue

```shell
import os
os.environ.setdefault("CUDA_VISIBLE_DEVICES", "-1")
os.environ.setdefault("TF_CPP_MIN_LOG_LEVEL", "3")

import tensorflow as tf

dtype = tf.float64

expected = [5.551115123125775e-17, -5.551115123125775e-17]

z = tf.Variable([37.42994775023705, 0.0], dtype=dtype)
cotangent = tf.constant([1.0, 0.0], dtype=dtype)

with tf.GradientTape() as tape:
    y = tf.nn.softmax(z, axis=-1)
    objective = tf.reduce_sum(y * cotangent)

actual = tape.gradient(objective, z)
vjp_sum = tf.reduce_sum(actual)

print("actual:")
print(actual.numpy().tolist())
print("expected:")
print(expected)
print("VJP sum:")
print(float(vjp_sum))

if actual.numpy()[0] == 0.0 and expected[0] != 0.0:
    print("BUG REPRODUCED: tf.nn.softmax loses a finite first-derivative signal")
else:
    print("not reproduced")
```

### Relevant log output

```shell
actual:
[0.0, -5.551115123125776e-17]
expected:
[5.551115123125775e-17, -5.551115123125775e-17]
VJP sum:
-5.551115123125776e-17
BUG REPRODUCED: tf.nn.softmax loses a finite first-derivative signal
```

### PR DESCRIPTION
**Pull request by GodlyDonuts**
Closes #126633.

Since this PR opened, #126748 landed the float64 softmax VJP correction on `master`. This rebased change preserves that implementation and adds an exact graph/eager regression for the saturated two-class input reported in #126633. The test verifies both the finite tail component and the zero-sum gradient invariant.

Validation:
- `python3 -m py_compile tensorflow/python/kernel_tests/nn_ops/softmax_op_test.py`
- `git diff --check origin/master...HEAD`

### CODE REVIEW

**Comment by GodlyDonuts** [comment]
@gemini-code-assist /review

**Review by dmiltr3 — DISMISSED** [review]

**Comment by GodlyDonuts** [comment]
@dmiltr3 It was referencing a nonexistent Bazel target before. I fixed it now. Please take a look.

**Review by dmiltr3 — DISMISSED** [review]
Thank you for the PR! We have verified the changes and they safely prevent float64 rounding residual losses. The patch has also successfully passed all internal presubmit tests (TAP). 

LGTM! I will now proceed to authorize this PR for final merge.

**Comment by nithyak0204** [comment]
Hi @GodlyDonuts, Can you please resolve the conflicts? Thank you!

**Comment by GodlyDonuts** [comment]
Should be fixed now!

**Review by dmiltr3 — CHANGES_REQUESTED** [review]
Hi @GodlyDonuts,

Thank you for the update. It looks like the `import/copybara` check failed on the latest commit. 

The issue is that adding `"//tensorflow/python/framework:test_lib"` to the `deps` list in `tensorflow/python/kernel_tests/nn_ops/BUILD` creates a conflict when syncing to our internal repository, as the necessary dependencies are already provided by the existing `"//tensorflow/python/platform:client_testlib"` target.

Could you please revert the addition of `"//tensorflow/python/framework:test_lib"` in the `BUILD` file? You should keep `"//tensorflow/python/eager:backprop"` and the Python import changes, as they are correct. 

For the Windows CI (`build-windows-x86 / build-and-test`) failure, the error is an unrelated pre-existing timeout/compilation flake (`slim_model_importer.cc`), so please ignore it.

We will proceed with the merge once the `BUILD` file is corrected. Thank you!

**Review by dmiltr3 — DISMISSED** [review]
Thank you for making the requested updates to the `BUILD` dependencies! I can confirm the `import/copybara` check successfully passes now, alongside all our internal integration presubmits. We are good to go! Approving the PR for merge.

**Comment by nithyak0204** [comment]
Hi @GodlyDonuts, Can you please resolve the conflicts? Thank you!

**Comment by GodlyDonuts** [comment]
@nithyak0204 Should be fixed now!

**Review by dmiltr3 — APPROVED** [review]
LGTM! Internal presubmit tests have passed successfully. Approving the PR.
